File: src/test/java/boj/problems/step8/No4948Test.java
package boj.problems.step8;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.StringReader;
import java.io.StringWriter;
import java.util.stream.Collectors;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class No4948Test {
    File path = new File(".");
    String input = path.getAbsolutePath() + "/src/test/java/boj/problems/step8/No4948_input.txt";
    String output = path.getAbsolutePath() + "/src/test/java/boj/problems/step8/No4948_output.txt";

    @Test
    @DisplayName("베르트랑 공준 테스트")
    void 베르트랑_공준_테스트() throws IOException {
        System.out.println("베르트랑 공준 : https://www.acmicpc.net/problem/4948");

        BufferedReader br_given = new BufferedReader(new FileReader(input));
        BufferedReader br_want = new BufferedReader(new FileReader(output));
        StringWriter sw = new StringWriter();
        BufferedWriter bw_got = new BufferedWriter(sw);

        //No4948.solve(br_given, bw_got);

        bw_got.close();

        BufferedReader br_got = new BufferedReader(new StringReader(sw.getBuffer().toString()));

        String got = br_got.lines().collect(Collectors.joining());
        String want = br_want.lines().collect(Collectors.joining());

        assertEquals(want, got);
    }
}

File: src/test/kotlin/boj/problems/No4948Test.kt
package boj.problems

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class No4948Test : StringSpec({
    "베르트랑 공준 : https://www.acmicpc.net/problem/4948" {
        val given = """
            1
            10
            13
            100
            1000
            10000
            100000
            0
        """.trimIndent()
        val expected = """
            1
            4
            3
            21
            135
            1033
            8392
        """.trimIndent()

        No4948().solve(given.reader().buffered()) shouldBe expected
    }
})
